### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by Drunk-Monk** [We-the-People-civ4col-mod/Mod#309]
Please could you create a notification for when a city has stopped resisting / rebelling.

It would be great to have some response or notification from the King when you ask for troops and after you win a colonial war, such as "I have decided to give you etc" would be useful.

Change the icon for buildings generating money to something more appropriate, like a dollar sign or gold coins rather than the yellow exclamation mark.

Is it possible to remove the pop-up asking if you want war, anytime that you land on the unit of another faction?

**Comment by ErikSandbraaten** [comment]
@raystuttgart Something for you to do :)

**Comment by raystuttgart** [comment]
About the 1st point:
Sure, I could do that. But I remember that in the past players were actually complaining that there were already to many messages. But if we all want it, I am ok with it and would implement it.

About the 2nd point:
I don't really understand what should be done.
Why should the king give anything to you after you win a war in the colonies?
What should the message say exactly?

About the 3rd point:
I am quite sure that all messages for generation of money used to have the "GOLD" icon in RaR.
Was there some change in the GameFonts considering the order of Icons?

About the 4th point:
The Pop-Up asking for war could be removed of course but are we sure we want that.
Because I usually only move my military units explicitly on a plot with enemy units if I really want to fight them. Otherwise I usually simply send them to their destination (in an automated way).
To sum it up - I am personally not really interested in removing that pop-up.

To sum it up:
1. I would do it if we **all** agree to it. I personally agree. @Nightinggale What about you?
2. I don't understand what is wanted. 
3. I first want to know how it could get broken. I hate messing in GameFonts when they are broken. It is extremely tedious.
4. I am personally not interested in that one.

**Comment by Nightinggale** [comment]
> I just saw that the GameFonts were modified (cleaned up) compared to RaR.
Some unused icons seem to have been removed.
Maybe however this introduced this Gold Icon bug (Point 3) which then could probably then also only be solved in the GameFonts and not in the DLL. (Might be wrong though. Just guessing.)

I did this to fix #164. The problem was that there are two gamefont readers, one for city billboards (the nameplate on the map) and one for python screens. Those two had gone out of sync because they treat unused icons differently. I wrote a bunch of debug display, which can be enabled from the domestic advisor, which I left in the game, but it will only show up if the game uses a debug dll. Perhaps it would be a good idea to document how to use this in order to... you know tell people that there is a tool to view GameFonts.

It's possible that I messed up something. It's not a huge issue if I deleted something we need because GameFontEditor can copy icons and paste them into another file. We still have the old GameFont files, both in releases and in git meaning we haven't actually lost graphics. It's just a matter of inserting the icons at the right spot.

**Comment by raystuttgart** [comment]
Hi guys I just checked the Notifications (on top of the Screen).

I also checked other mouse over messages and also so the correct Icon.

For me the correct gold icon is displayed. (It is the stack of chips that Schmiddie had created)

@Nightinggale 
The GameFonts seem to work nicely.
And yes, they are much cleaner now. Great Job.

So let us please not revert an Icon that Schmiddie had put effort into creating.
That feels very impolite to do. And actually I like that Icon as well.

So considering the "Gold Icon" there is no fix needed for me.

**Comment by Drunk-Monk** [comment]
About point 2 right now when after you ask the King for troops the chat window abruptly closes, I would just like to see a message saying "Very well I am dispatching reinforcements" or something similar. 

Similarly when War ends there is no acknowledgement, I would like to see that the King thanks you and a boost in relations, you can destroy the all of the oppositions colonies but there is no recognition of this. 

About point 4 its possible to be surrounded by Native Nations, who have troops running all over your land. It gets tedious having to deal with the possibility of declaring war by accident, every time you move a unit.

**Comment by raystuttgart** [comment]
Point 1: "End Uprest message" should be easy to do and will be implemented if nobody minds more messages.
Point 2: I will check how much effort it is. I hardly remember my DLL-Diplo-Events. Too long ago.
----------
Point 3: The Icon is displayed perfectly. Don't see any need for a fix.
Point 4: I would really also rather leave it as it is. This could also have side effects if removed.

**Comment by Nightinggale** [comment]
> Point 1: "End Uprest message" should be easy to do and will be implemented if nobody minds more messages.

I would say add it. It's not something, which will spam considering how rare the message should be.
As for getting way to many messages, I think we should give the message system an overhaul to make it easier for the player to control how to be notified. However that can wait.

> Point 4: I would really also rather leave it as it is. This could also have side effects if removed.

Me too, though it would be ok to add "don't ask again this turn" or something like that. The question is how hard that would be to implement and if it's worth the time to do so and if it is, when would it be fine (relative to other tasks).

**Comment by raystuttgart** [comment]
Point 1: "End Uprest message" is implemented now and tested.
I will push it to "fixes".

**Edit:**
1. Since I did not want to mess with current XML and the current issue we currently have with missing TXT_KEYs I have created **a new XML CIV4GameText_WTP.xml** for this message text. Once the mentioned issue is fixed the entry should of course **later be integrated** in one of the other XMLs.
2. I am not able to write **French, Spanish and Italian** texts (only English and Germany).
It would be good if somebody speaking those languages could write appropriate **translation** texts.

**Comment by Nightinggale** [comment]
If you include UTF8 in the name of a text xml file, then the game will assume the file to be utf8 encoded. If not, then it will assume it to be encoded in windows 1252. Utf8 means you can write characters like ä directly without using html escapes.

I wrote it like this to ensure that the system is backward compatible and is still able to read vanilla files.

**Comment by raystuttgart** [comment]
So I should rename the file with UTF8 and push again?
The current file worked flawlessly for me during my tests though.

It is supposed to be only a temporarily file anyways until our lost **TXT_Keys issue** is fixed.
I think we definitely need to fix this for the next public release by the way.

**Comment by Nightinggale** [comment]
Ideally we should have UTF8 in all the filenames as it helps when adding non-ASCII characters. For all I know it will make a difference in French or it will make a difference in the next string added to the file. However now that it's there and it's meant to be changed prior to next release anyway, then it's not a big deal.

We should also consider how to organize the strings. A category of strings called GameText isn't great because it quickly ends up as a mix of strings from all sorts of locations. It's not really possible to tell which file contains a string in question when they are all some sort of GameText. The new files named units, buildings etc contains all the strings the name indicates and nothing else.

Another issue is that you added English in non-English languages. I changed how strings are loaded. Instead of using index like vanilla, it now uses the tag. This means if you select German, you will get the line with the German tag regardless of order. Also if your chosen language isn't present, then it will use English for the string in question. This means if you only have English and German, then only add those two. We can script find untranslated lines if they are missing, but it's a bit more work to teach the script to locate English text in non-English languages, particularly if the English string updates, but it doesn't in the other languages (it has happened), in which case a simple string comparison won't work.

> It is supposed to be only a temporarily file anyways until our lost TXT_Keys issue is fixed.

And we really should look into this. I'm not sure how it manage to break itself, but apparently there is at least one more issue to look into.

**Comment by raystuttgart** [comment]
I will close this for now.

1. is fixed
2. we will need to rework DLL Diplo Events anyway later
3. Icon works (small stash of gold coins)
4. Ask for War-Popup I will rather not remove

Also the XML file is now renamed with "UTF8" at the end and all special characters work nicely.

Also the "lost TXT_KEYs" is another issue that is in work anyways.
